content/docs/what-is-kudo.md

```diff
@@ -138,6 +138,19 @@ $ kubectl kudo install kafka --instance dev-kafka
 
 ### Limitations
 
+#### Cross-namespace ownership and cluster-scoped resources
+
 All the resources created by an operator _Instance_ are [owned](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#owners-and-dependents) by it. This way, when the _Instance_ is deleted, the resources are removed too. However, the _Instance_ (as well as  _Operator_, and _OperatorVersion_)  are namespace scoped. Namespaced resources can only own resources in the same namespace. As a result, it isn't possible for a single operator to create resources in multiple namespaces.
 
 It is possible for an _Instance_ to create cluster-scoped resources. However, these `resourceOwners` field for such resources is not populated with the _Instance_ reference. Cluster-scoped resources are also **not** automatically cleaned up when the _Instance_ is deleted. Additionally,  they can result in issues when updating or upgrading an _Instance_. [KEP-5](https://github.com/kudobuilder/kudo/blob/master/keps/0005-cluster-resources-for-crds.md) will resolve this limitation.
+
+#### Watching arbitrary resources
+
+KUDO cannot watch arbitrary user-specified resource types.
+
+The only things that KUDO controller watches are changes to the `spec` KUDO-specific
+`Instance`, `OperatorVersion` and `Operator` custom resources. In reaction it runs
+[plans](#operator-plans) which can deploy almost arbitrary other resources.
+However, other than checking their health and re-applying until the health checks pass,
+KUDO never observes their status once the plan completes.
```

Comment:
That is a bit misleading because KUDO in fact watches for changes of _some_ resource types. But the controllers then ensure that O/OV/I is deployed "correctly", i.e. the resources of the current plan/step/phase are applied on the cluster. That's, IMO, the more general limitation here: The lifecycle of operators managed by KUDO have to fit into the plan/step/phase state machine provided by KUDO. Other models aren't supported. E.g., the MySQL scenario mentioned in Slack, where a Master pod failure should result in a role switch of a new pod doesn't fit into this model.

Reply:
> KUDO in fact watches for changes of some resource types

Can you please provide a link to the code or example? I'd like to understand this better.

Reply:
It's this part: https://github.com/kudobuilder/kudo/blob/master/pkg/controller/instance/instance_controller.go#L97
It tells Kubernetes that this controller is managing `Instance` resources but wants to get reconciliation notifications if certain resources owned by the `Instance` change their state. This is done so that the controller can react to certain events that might affect the operator's plan/step/phase state machine.